File: RepoTransBench/target_projects/Java/Go/scottyab_AESCrypt-Android/public_tests/application_public_test.go
package public_tests

import (
	"testing"
)

func TestEncryptDecryptPublic(t *testing.T) {
	password := "anotherSecret"
	message := "public test string!"

	SetDebugLog(true)

	encryptedMsg, err := Encrypt(password, message)
	if err != nil {
		t.Fatalf("error occurred during encrypt in public test: %v", err)
	}

	messageAfterDecrypt, err := Decrypt(password, encryptedMsg)
	if err != nil {
		t.Fatalf("error occurred during decrypt in public test: %v", err)
	}

	if messageAfterDecrypt != message {
		t.Errorf("messages don't match after encrypt and decrypt in public test: got %q, want %q", messageAfterDecrypt, message)
	}
}

func TestEncryptPublic(t *testing.T) {
	password := "publicPass"
	message := "anotherMessage"
	encryptedMsg, err := Encrypt(password, message)
	if err != nil {
		t.Fatalf("error occurred during encrypt in public test: %v", err)
	}
	if encryptedMsg == "" {
		t.Error("encryptedMsg should not be empty")
	}
}

func TestDecryptPublic(t *testing.T) {
	password := "somePassword"
	encryptedMsg := "i5OSk38FnX6OGv5CeXf2iA=="
	result, err := Decrypt(password, encryptedMsg)
	if err != nil {
		t.Fatalf("error occurred during decrypt in public test: %v", err)
	}
	if result != "testOne" {
		t.Errorf("expected 'testOne', got %q", result)
	}
}

File: RepoTransBench/target_projects/Java/C#/scottyab_AESCrypt-Android/public_tests/ApplicationPublicTest.cs
using System;
using Xunit;
using AESCryptLib;
using System.Security.Cryptography;

namespace AESCryptPublicTests
{
    public class ApplicationPublicTest
    {
        [Fact]
        public void TestEncryptDecryptPublic()
        {
            string password = "anotherSecret";
            string message = "public test string!";

            // Set debug enabled if needed
            AESCrypt.DEBUG_LOG_ENABLED = true;

            string encryptedMsg = null;
            try
            {
                encryptedMsg = AESCrypt.Encrypt(password, message);
            }
            catch (CryptographicException)
            {
                Assert.True(false, "error occurred during encrypt in public test");
            }
            catch (ArgumentNullException)
            {
                Assert.True(false, "null param during encrypt in public test");
            }

            string messageAfterDecrypt = null;
            try
            {
                messageAfterDecrypt = AESCrypt.Decrypt(password, encryptedMsg);
            }
            catch (CryptographicException)
            {
                Assert.True(false, "error occurred during decrypt in public test");
            }
            catch (ArgumentNullException)
            {
                Assert.True(false, "null param during decrypt in public test");
            }

            Assert.Equal(message, messageAfterDecrypt);
        }

        [Fact]
        public void TestEncryptPublic()
        {
            string password = "publicPass";
            string message = "anotherMessage";

            try
            {
                string encryptedMsg = AESCrypt.Encrypt(password, message);
                Assert.NotNull(encryptedMsg);
            }
            catch (CryptographicException)
            {
                Assert.True(false, "error occurred during encrypt in public test");
            }
            catch (ArgumentNullException)
            {
                Assert.True(false, "null param during encrypt in public test");
            }
        }

        [Fact]
        public void TestDecryptPublic()
        {
            string password = "somePassword";
            string encryptedMsg = "i5OSk38FnX6OGv5CeXf2iA==";
            try
            {
                string result = AESCrypt.Decrypt(password, encryptedMsg);
                Assert.Equal("testOne", result);
            }
            catch (CryptographicException)
            {
                Assert.True(false, "error occurred during decrypt in public test");
            }
            catch (ArgumentNullException)
            {
                Assert.True(false, "null param during decrypt in public test");
            }
        }
    }
}